extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -102,11 +50,55 @@ public function renderInfobox( array $infoboxdata, $theme, $layout ) {
 			$output = '';
 		}
 
-		\Wikia\PortableInfobox\Helpers\PortableInfoboxDataBag::getInstance()->setFirstInfoboxAlredyRendered( true );
+		return $output;
+	}
 
-		wfProfileOut( __METHOD__ );
+	protected function getRenderHelper() {
+		if ( !isset( $this->helper ) ) {
+			$this->helper = new PortableInfoboxRenderServiceHelper();
+		}
+		return $this->helper;
+	}
 
-		return $output;
+	/**
+	 * Produces HTML output for item type and data
+	 *
+	 * @param $type
+	 * @param array $data
+	 * @return string
+	 */
+	protected function render( $type, array $data ) {
+		return $this->templateEngine->render( $type, $data );
+	}
+
+	/**
+	 * renders part of infobox
+	 *
+	 * @param string $type
+	 * @param array $data
+	 *
+	 * @return string - HTML
+	 */
+	protected function renderItem( $type, array $data ) {
+		switch ( $type ) {
+			case 'group':
+				$result = $this->renderGroup( $data );
+				break;
+			case 'header':
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "header" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)